instrumentation/src/main.c

```diff
@@ -0,0 +1,101 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#define MAX_LINE_LENGTH 1023
+#define MAX_LINES 50
+
+// TODO change to systemd drop in file paths
+static char *const env_var_file_java = "/etc/splunk/zeroconfig_java.conf";
+static char *const env_var_file_node = "/etc/splunk/zeroconfig_node.conf";
+
+static char *const allowed_env_vars[] = {
+    "OTEL_SERVICE_NAME",
+    "OTEL_EXPORTER_OTLP_ENDPOINT",
+    "OTEL_RESOURCE_ATTRIBUTES",
+    "SPLUNK_PROFILER_ENABLED",
+    "SPLUNK_PROFILER_MEMORY_ENABLED",
+    "SPLUNK_METRICS_ENABLED",
```

Comment:
this list needs to be completed. It doesn't represent the whole set yet.

Reply:
For java, would we need to support all of the [splunk-specific env vars](https://docs.splunk.com/Observability/gdi/get-data-in/application/java/configuration/advanced-java-otel-configuration.html)?

Any from [otel java](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#configuring-with-environment-variables) and [otel sdk](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/) not included in the above list?

Reply:
yes we're missing quite a few.

Reply:
Quick question for clarification: Correct me if I am wrong, wouldn't the allowed env variables be injected from the language specific drop-in files and not the constant ones defined here?

Reply:
This allows us to check the file contents, and ensure that only the keys we allow here are injected in the environment of processes.

Reply:
Will this library read `JAVA_TOOL_OPTIONS` and `NODE_OPTIONS` from the conf files, or are you expecting them to be defined somewhere else?  For systemd, we can define them directly in the drop-in file(s), but should they be added to the list for the preloader method?

Reply:
If we are allowing arbitrary input I think instrumentation checksum validation from .rodata should be required. This will add complexity and overhead but the benefit is that we'd only respect known target values.